jbpm-flow/src/main/java/org/jbpm/process/core/async/AsyncSignalEventCommand.java

```diff
@@ -48,10 +48,9 @@ public ExecutionResults execute(CommandContext ctx) throws Exception {
         if (runtimeManager == null) {
             throw new IllegalArgumentException("No runtime manager found for deployment id " + deploymentId);  
         }
-        RuntimeEngine engine = runtimeManager.getRuntimeEngine(ProcessInstanceIdContext.get(processInstanceId));        
+        RuntimeEngine engine = runtimeManager.getRuntimeEngine(ProcessInstanceIdContext.get(processInstanceId.longValue()));        
```

Comment:
If the context doesn't contain processInstanceId, it will throw a NPE.
Could the code be protected including also the processInstanceId as required, same as deploymentId and signal in the previous "if" comparison, wdyt?

Reply:
done